contracts/lease/src/position/mod.rs

```diff
@@ -781,3 +816,94 @@ mod test_check {
         Position::new(asset, liability, min_asset, min_sell_asset)
     }
 }
+
+#[cfg(test)]
+mod test_validate_close {
+    use currency::{lpn::Usdc, test::Dai};
+    use finance::{
+        coin::Coin,
+        duration::Duration,
+        liability::Liability,
+        percent::Percent,
+        price::{self, Price},
+    };
+
+    use crate::error::ContractError;
+
+    use super::Position;
+
+    type TestCurrency = Dai;
+    type TestLpn = Usdc;
+
+    #[test]
+    fn too_small_amount() {
+        let spec = position(100.into(), 75.into(), 15.into());
+        let result = spec.validate_close_amount(14.into(), price(1.into(), 1.into()));
+        assert!(matches!(
+            result,
+            Err(ContractError::PositionCloseAmountTooSmall(_))
+        ));
+    }
+
+    #[test]
+    fn amount_as_min_sell_asset() {
+        let spec = position(100.into(), 75.into(), 15.into());
+        let result = spec.validate_close_amount(15.into(), price(1.into(), 1.into()));
+        assert!(result.is_ok())
+    }
+
+    #[test]
+    fn too_big_amount() {
+        let spec = position(100.into(), 25.into(), 1.into());
+        let result = spec.validate_close_amount(76.into(), price(1.into(), 1.into()));
+        assert!(matches!(
+            result,
+            Err(ContractError::PositionCloseAmountTooBig(_))
+        ));
+    }
+
+    #[test]
+    fn amount_as_min_asset() {
+        let spec = position(100.into(), 25.into(), 1.into());
+        let result = spec.validate_close_amount(75.into(), price(1.into(), 1.into()));
+        assert!(matches!(
+            result,
+            Err(ContractError::PositionCloseAmountTooBig(_))
+        ));
+    }
+
+    #[test]
+    fn valid_amount() {
+        let spec = position(100.into(), 40.into(), 15.into());
+        let result = spec.validate_close_amount(53.into(), price(1.into(), 1.into()));
+        assert!(result.is_ok())
+    }
+
+    fn position(
+        amount: Coin<TestCurrency>,
+        min_asset: Coin<TestLpn>,
+        min_sell_asset: Coin<TestLpn>,
```

Comment:
introduce type arguments that are `: Into<Currency<Xyz>>` to simplify the calling, e.g. `position(100, 25, 1)`